src/modules/terraform.rs

```diff
@@ -25,20 +27,42 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     let mut module = context.new_module("terraform");
     let config: TerraformConfig = TerraformConfig::try_load(module.config);
 
-    module.set_style(config.style);
-    module.create_segment("symbol", &config.symbol);
+    let terraform_version =
+        format_terraform_version(&utils::exec_cmd("terraform", &["version"])?.stdout.as_str());
+    let terraform_workspace = get_terraform_workspace(&context.current_dir);
```

Comment:
Well, since calculating terraform version is very slow on large workspaces, you'd better move this to the `.map` function so that it will be calculated only when the formatter saw a `$version` variable.

Also users should be informed about this in the doc.